src/hooks/useHandlers.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
+import { useActions } from './useActions';
+import { useTypedSelector } from './useTypedSelector';
+
+import { IPureTodo } from '../models/ITodo';
+import { ITodo } from '../models/ITodo';
+
+const defaultValue = {
+  id: '',
+  title: '',
+  description: '',
+  status: '',
+};
+
+export const useHandlers = () => {
+  const {
+    addTodoActionCreator,
+    deleteTodoActionCreator,
+    updateTodoStatusActionCreator,
+    updateTodoActionCreator,
+    setVisible,
+    setType,
+  } = useActions();
+  const { todos } = useTypedSelector((state) => state.todosReducer);
+  const [initialValue, setInitialValue] = useState(defaultValue as ITodo);
+
+  return {
+    addTodo: async (todo: IPureTodo) => {
+      addTodoActionCreator(todo);
+      setVisible(false);
+    },
+
+    deleteTodo: async (id: string) => {
+      const currentTodo = todos.find((todo) => todo.id === id);
+      deleteTodoActionCreator(currentTodo!);
+    },
+
+    handleTodoProgress: async (id: string, status: string) => {
+      const currentTodo = todos.find((todo) => todo.id === id);
+      updateTodoStatusActionCreator({ ...currentTodo, status: status });
+    },
+
+    editTodo: async (todo: IPureTodo) => {
+      const { id } = initialValue;
+      const currTodo = todos.find((todo) => todo.id === id);
+      const editedTodo = {
+        ...currTodo!,
+        ...todo,
+      };
+      updateTodoActionCreator(editedTodo);
+      setVisible(false);
+    },
+
```

Comment:
I see errors
![image](https://user-images.githubusercontent.com/40803254/171046880-ba8d709a-eb70-420c-b013-cd99cc81d0e8.png)

Reply:
Sometimes after changing branch, i have the same, but after saving the file manually once again, it works... weird situation, dunno why it happens..